plugins/inputs/mikrotik/mikrotik.go

```diff
@@ -0,0 +1,227 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mikrotik
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io"
+	"net/http"
+	"time"
+
+	common_tls "github.com/influxdata/telegraf/plugins/common/tls"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Mikrotik struct {
+	Address         string          `toml:"address"`
+	IgnoreCert      bool            `toml:"ignore_cert,omitempty"`
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+	IgnoreComments  []string        `toml:"ignore_comments"`
+	IncludeModules  []string        `toml:"include_modules"`
+	Username        string          `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	Log             telegraf.Logger `toml:"-"`
+	common_tls.ClientConfig
+
+	tags map[string]string
+
+	url []mikrotikEndpoint
+
+	client *http.Client
+
+	systemTagsURL []string
+}
+
+func (h *Mikrotik) Start() error {
+	return h.getSystemTags()
+}
+
+func (*Mikrotik) SampleConfig() string {
+	return sampleConfig
+}
+
+func (h *Mikrotik) Init() error {
+	if h.Username == "" {
+		return wrapError("mikrotik init", nil)
```

Comment:
Please just use `fmt.Errorf` here directly and use a more speaking error message. Please remember, these messages are user-facing! So maybe
```suggestion
		return errors.New("username must be set")
```